server/plugins/authentication.js

```diff
@@ -0,0 +1,158 @@
+const { makeExtendSchemaPlugin, gql } = require("graphile-utils");
+
+const PassportLoginPlugin = makeExtendSchemaPlugin(build => ({
+  typeDefs: gql`
+    input RegisterInput {
+      username: String!
+      email: String!
+      password: String!
+      name: String
+      avatarUrl: String
+    }
+
+    type RegisterPayload {
+      user: User! @pgField
+    }
+
+    input LoginInput {
+      username: String!
+      password: String!
+    }
+
+    type LoginPayload {
+      user: User! @pgField
+    }
+
+    extend type Mutation {
+      register(input: RegisterInput!): RegisterPayload
+      login(input: LoginInput!): LoginPayload
+    }
+  `,
+  resolvers: {
+    Mutation: {
+      async register(
```

Comment:
Is it possible to add an Comment here which then shows up in GraphiQL that login/register is a custom resolver? It just took me ages to hunt it down to this file?

Reply:
Absolutely; you can add it just above the `register` field itself using standard GraphQL syntax:

```graphql
extend type Mutation {
   """
   Documentation goes here
   """
   register(input: RegisterInput!): RegisterPayload
```